rewrite-gradle/src/main/java/org/openrewrite/gradle/trait/GradleDependency.java

```diff
@@ -17,64 +17,1291 @@
+     * For platform dependencies, automatically looks at the inner dependency.
+     *
+     * @return The declared version string (literal or variable name), or null if it cannot be determined
+     */
+    public @Nullable String getDeclaredVersion() {
+        // For version variables, getVersionVariable() already handles this
+        String versionVar = getVersionVariable();
+        if (versionVar != null) {
+            return versionVar;
+        }
+
+        // If this is a platform dependency, delegate to the inner dependency
+        if (isPlatform()) {
+            GradleDependency platformDep = getPlatformDependency();
+            return platformDep != null ? platformDep.getDeclaredVersion() : null;
+        }
+
+        J.MethodInvocation m = cursor.getValue();
+        List<Expression> depArgs = m.getArguments();
+
+        if (depArgs.isEmpty()) {
+            return null;
+        }
+
+        Expression arg = depArgs.get(0);
+
+        // String literal notation: "group:artifact:version"
+        if (arg instanceof J.Literal && ((J.Literal) arg).getValue() instanceof String) {
+            Dependency dep =
+                    Dependency.parse((String) ((J.Literal) arg).getValue());
+            return dep != null ? dep.getVersion() : null;
+        }
+
+        // Map notation - look for "version" entry
+        if (depArgs.size() >= 3) {
+            for (Expression e : depArgs) {
+                if (e instanceof G.MapEntry) {
+                    G.MapEntry entry = (G.MapEntry) e;
+                    if (entry.getKey() instanceof J.Literal &&
+                            "version".equals(((J.Literal) entry.getKey()).getValue())) {
+                        return extractValueAsString(entry.getValue());
+                    }
+                } else if (e instanceof J.Assignment) {
+                    J.Assignment assignment = (J.Assignment) e;
+                    if (assignment.getVariable() instanceof J.Identifier &&
+                            "version".equals(((J.Identifier) assignment.getVariable()).getSimpleName())) {
+                        return extractValueAsString(assignment.getAssignment());
+                    }
+                }
+            }
+        }
+
+        // Map literal notation
+        if (arg instanceof G.MapLiteral) {
+            for (G.MapEntry entry : ((G.MapLiteral) arg).getElements()) {
+                if (entry.getKey() instanceof J.Literal &&
+                        "version".equals(((J.Literal) entry.getKey()).getValue())) {
+                    return extractValueAsString(entry.getValue());
+                }
+            }
+        }
+
+        return null;
+    }
+
+    /**
+     * Helper method to extract a string value from various expression types.
+     * Handles literals, identifiers, field access, method invocations, and GStrings.
+     */
+    private @Nullable String extractValueAsString(Expression value) {
```

Comment:
I'm a little worried about the prints contained in here for a couple of reasons:

* Sustainability/Incorrect results
* If a marker exists on one of these elements from a previous recipe, then the marker information will end up being printed here

Reply:
I've added in a `withMarkers(Markers.EMPTY)` to handle the marker problem. Otherwise leaving it mostly the same for now... agree there's possibility for error in there but it's just pretty convenient to be able to get a string out of whatever format the expression is.